src/Package.zig

```diff
@@ -510,6 +510,18 @@ fn fetchAndUnpack(
             // I have not checked what buffer sizes the xz decompression implementation uses
             // by default, so the same logic applies for buffering the reader as for gzip.
             try unpackTarball(gpa, &req, tmp_directory.handle, std.compress.xz);
+        } else if (ascii.eqlIgnoreCase(content_type, "application/octet-stream")) {
+            // TODO more properly parse the content-disposition header according to https://www.ietf.org/rfc/rfc2183.txt.
+            //      the following is a minimal check as i'm not sure where content-disposition parsing should happen.
+            // support gitlab tarball urls such as https://gitlab.com/<namespace>/<project>/-/archive/<sha>/<project>-<sha>.tar.gz
+            // whose content-disposition header is: 'attachment; filename="<project>-<sha>.tar.gz"'
+            const content_disposition = req.response.headers.getFirstValue("Content-Disposition") orelse
+                return report.fail(dep.url_tok, "Missing 'Content-Disposition' header for Content-Type=application/octet-stream", .{});
+            if (ascii.indexOfIgnoreCasePos(content_disposition, 0, "attachment") != null and
+                ascii.indexOfIgnoreCasePos(content_disposition, 0, ".tar.gz") != null)
```

Comment:
Use mem.startsWith, mem.endsWith, lose the TODO comment, and this is good to go.

Reply:
Thanks! Done.